src/utils/redstone_tx_sync/client.go

```diff
@@ -0,0 +1,128 @@
+package redstone_tx_sync
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/sirupsen/logrus"
+	"github.com/warp-contracts/syncer/src/utils/bundlr"
+	"github.com/warp-contracts/syncer/src/utils/config"
+	"github.com/warp-contracts/syncer/src/utils/logger"
+	"github.com/warp-contracts/syncer/src/utils/model"
+	"github.com/warp-contracts/syncer/src/utils/sequencer"
+	sequencer_types "github.com/warp-contracts/syncer/src/utils/sequencer/types"
+	"gorm.io/gorm"
+)
+
+type Client struct {
+	db              *gorm.DB
+	Log             *logrus.Entry
+	ethClient       *ethclient.Client
+	sequencerClient *sequencer.Client
+}
+
+func NewClient(config *config.Config) (self *Client) {
+	self = new(Client)
+	self.Log = logger.NewSublogger("redstone_tx_sync_client")
+
+	return
+}
+
+func (self *Client) WithEthClient(url EthUrl) *Client {
+	ethClient, err := GetEthClient(self.Log, url)
+	if err != nil {
+		self.Log.WithError(err).Error("Could not get ETH client")
```

Comment:
initialize client elsewhere

Reply:
IMHO it's a good practice in commenting PRs to not only write 'what', but also 'why'...and propose concrete solutions (i.e. in this case - 'where')

Reply:
ok. Initialize client in the Controller